### PR TITLE
Animation optimized

### PR DESCRIPTION
**Pull request by Mumsfilibaba**
* Added PrehashedString - Takes a string and computes a hash once, if not changed the hash is not recomputed. 
* Interpolation in animation system is improved'
* Parallelization in skinning compute shader is improved

### CODE REVIEW

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 1]
```diff
@@ -712,7 +720,7 @@ namespace LambdaEngine
 						memcpy(pMappedWeights, pMesh->VertexBoneData.GetData(), vertexWeightBufferDesc.SizeInBytes);
 						pVertexWeightStagingBuffer->Unmap();
 
-						vertexWeightBufferDesc.DebugName	= "Vertex Weight Buffer";
+						vertexWeightBufferDesc.DebugName	= "Vertex VertexWeight Buffer";
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]:
```suggestion
						vertexWeightBufferDesc.DebugName	= "Vertex Weight Buffer";
```

File: LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp [hunk 2]
```diff
@@ -698,7 +706,7 @@ namespace LambdaEngine
 						VALIDATE(meshEntry.pAnimatedVertexBuffer != nullptr);
 
 						BufferDesc vertexWeightBufferDesc = {};
-						vertexWeightBufferDesc.DebugName	= "Vertex Weight Staging Buffer";
+						vertexWeightBufferDesc.DebugName	= "Vertex VertexWeight Staging Buffer";
```
Comment (skumbananaaa) [inline LambdaEngine/Source/Game/ECS/Systems/Rendering/RenderSystem.cpp]:
```suggestion
						vertexWeightBufferDesc.DebugName	= "Vertex Weight Staging Buffer";
```

File: LambdaEngine/Include/Containers/PrehashedString.h [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+			m_Str			= std::move(other.m_Str);
+			m_Hash			= CalculateHash();
+			other.m_Hash	= 0;
+			return *this;
+		}
+
+		inline operator String() const
+		{
+			return m_Str;
+		}
```
Comment (skumbananaaa) [inline LambdaEngine/Include/Containers/PrehashedString.h:109]: Return as reference, const reference?

Comment (Mumsfilibaba) [reply LambdaEngine/Include/Containers/PrehashedString.h:109]: Added member functions for this. This operator will instead be used when we need to convert the whole object into a string.

**Review by skumbananaaa — APPROVED** [review]

**Pull request merged**